strawberryfields/backends/base.py

```diff
@@ -478,6 +479,24 @@ def measure_homodyne(self, phi, mode, select=None, **kwargs):
         Returns:
             float: measured value
         """
+        print("HI")
+        raise NotImplementedError
+          
+    
+    def measure_fock(self, modes, shots=1, select=None, **kwargs):
+        """Measure the given modes in the Fock basis.
+
+        Updates the current state of the circuit to the conditional state of this measurement result.
+
```

Comment:
```suggestion
    If ``shots`` > 1, the system state is collapsed to the state corresponding to the last measurement sample.
```

Reply:
And the same addition for the other measure_* methods as well.

Reply:
@nquesada and I were talking about the same thing today. My preference for the moment is to disallow `shots` for measurements that don't appear at the end of a circuit. I think it could confuse a lot of people if we were to update the conditional state in this (somewhat arbitrary) way

Reply:
What if the user uses `shots>1` and requests a state object from LocalEngine?

Reply:
For the moment, we are only going to support `shots` in a very narrow scope: for `measure_fock` on the Gaussian backend. More subtle issues that come up in the general case will still have to be worked out.

Reply:
In this narrow case, there is no relevant conditional state anyway (gaussian backend does not allow for non-Gaussian conditional states)

Reply:
For that case, there are three options:
i) reset all modes to vacuum
ii) implement some kind of "block" where gaussian backend refuses any new ops post-fock-measurements
iii) keep the same pre-measurement Gaussian state, even after photon measurement

I'm currently leaning towards iii, since it is not too surprising and allows users to keep using the same circuit

Reply:
>  ii) implement some kind of "block" where gaussian backend refuses any new ops post-fock-measurements

I quite like option (ii), as it can very naturally be done in the front end validation/compilation step

Reply:
> @nquesada and I were talking about the same thing today. My preference for the moment is to disallow shots for measurements that don't appear at the end of a circuit. I think it could confuse a lot of people if we were to update the conditional state in this (somewhat arbitrary) way

thumb

Reply:
I'm in favor of option (ii) too. If the block is in place, the `backend.state()` method could just raise an error.

The measure_* docstrings in strawberryfields/backends/base.py still need to be updated to reflect what is actually happening, though :)

Reply:
Let's move this discussion to a future issue. For the sake of this PR, we will do the following:
1. Gaussian backend will only implement `shots` for `measure_fock`. The other Gaussian measurements will give a `NotImplementedError` if `shots > 1`. I will remove the relevant code in the other measurements for the moment (not sure I trust it)
1. Gaussian backend will not update conditional state if `shots > 1`. I will add code to give a warning that the conditional state is not updated. This is a simple replacement for the above option (ii) for the moment, which we can think about implementing in a future PR.
1. Support for postselection and shots should be easy to add, but I will leave for future PR (for discussion: we might want to allow user to post-select on subset of modes)
1. Other backends will raise `NotImplementedError` for every measurement type if `shots > 1`
1. I will update docstrings accordingly